recipes/cpprestsdk/all/conanfile.py

```diff
@@ -46,17 +46,20 @@ def configure(self):
             del self.options.fPIC
 
     def requirements(self):
-        self.requires("openssl/1.1.1f")
+        self.requires("openssl/1.1.1g")
         if self.options.with_compression:
             self.requires("zlib/1.2.11")
         if self.options.with_websockets:
-            self.requires("websocketpp/0.8.1")
+            self.requires("websocketpp/0.8.2")
         self.requires("boost/1.72.0")
 
     def source(self):
         tools.get(**self.conan_data["sources"][self.version])
         extracted_dir = self.name + "-" + self.version
         os.rename(extracted_dir, self._source_subfolder)
+        if "patches" in self.conan_data and self.version in self.conan_data["patches"]:
```

Comment:
do not apply patches under source folder. Keep it on build folder. Why? Any change in source folder will require a new download, when developing a new recipe it will consume a lot of extra time for every new change.

Reply:
Huh 😕 very interesting. In my head the source command should do all the handling of the sources. That's a very good reason.

How does conan know the sources do not match the download?

Reply:
each version is downloaded in a separated folder. Also, conan copies the source folder content to build folder, thus applying patch in build is better, because for every build, conan removes the build folder, not the source folder.

Reply:
Ahh. thank you for the insights. I think I made this mistake in other recipes.